### PR TITLE
Add total diff field to root block

### PR DESCRIPTION
**Pull request by ninjaahhh**
so we can update root chain's fork choice rule to be based on total difficulty

cc @pingke

### CODE REVIEW

File: quarkchain/core.py [hunk 1]
```diff
@@ -856,6 +858,7 @@ class RootBlockHeader(Serializable):
         ("extra_data", PrependedSizeBytesSerializer(2)),
         ("mixhash", hash256),
         ("signature", FixedSizeBytesSerializer(65)),
+        ("total_difficulty", biguint),
```
Comment (qizhou) [inline quarkchain/core.py]: I would suggest moving total_difficulty field just after difficulty field - in general, we want to leave signature to be the last field in the header.

Comment (qizhou) [inline quarkchain/core.py]: I would suggest moving total_difficulty field just after difficulty field - in general, we want to leave signature to be the last field in the header.

Comment (ninjaahhh) [reply quarkchain/core.py]: done

Comment (ninjaahhh) [reply quarkchain/core.py]: done

**Pull request merged**